app/src/main/java/com/example/drawingactivity/ForcableEditText.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.widget.EditText
+
+@SuppressLint("ClickableViewAccessibility")
+class ForcableEditText @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : EditText(context, attrs, defStyle) {
+
+    override fun addTextChangedListener(watcher: TextWatcher?) {
+        watcher?.let { if (!textListeners.contains(it)) textListeners.add(it) }
+        super.addTextChangedListener(watcher)
+    }
+
+    private var textListeners = mutableListOf<TextWatcher>()
+
+    private fun removeAllListeners() {
+        textListeners.forEach { removeTextChangedListener(it) }
+    }
+
+    private fun addBackAllListenters() {
+        textListeners.forEach { addTextChangedListener(it) }
+    }
+
+    fun forceText(s: String) {
+        removeAllListeners()
+        if(!this.hasFocus())setText(s)
```

Comment:
If this has focus do you want to remove and all the listeners back? Wrap if statement around all of these?

Reply:
wrapping the whole block with the if